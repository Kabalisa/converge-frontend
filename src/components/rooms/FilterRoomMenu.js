import React, { Component } from 'react';
import { IconMenu } from 'react-toolbox/lib/menu';
import { Input, SelectInput as Select } from '../commons';
import '../../assets/styles/filterbutton.scss';

class FilterButton extends Component {
  state = {
    roomCapacity: 0,
    officeList: [],
    office: '',
    location: '',
    locationList: [],
  };
  handleInputChange = (e) => {
    e.preventDefault();
    // logic goes here
  };
  render() {
    const {
      roomCapacity, officeList, office, location, locationList,
    } = this.state;
    const filterIcon = () => (
      <div className="filterBtn">
        <span>Filter</span>
      </div>
    );
    return (
      <IconMenu
        className="filter-dropdown"
        icon={filterIcon()}
      >
        <div className="filter-search">
          <Input
            id="filterSearch"
            name="filterSearch"
            type="text"
            value=""
            labelName=""
            inputClass="mrm-input filter-search-input"
            placeholder="Search"
            onChange={this.handleInputChange}
            required
          />
        </div>
        <div className="other-filters">
          <Select
            labelText="Location"
            name="locationFilter"
            id="location-filter"
            value={location}
            options={locationList}
            onChange={this.handleInputChange}
            wrapperClassName="location-wrapper"
            placeholder="Select location"
            selectInputClassName="location-input default-select"
            required
          />
          <Select
            labelText="Office"
            name="OfficeFilter"
            id="office-filter"
            value={office}
            options={officeList}
            onChange={this.handleInputChange}
            wrapperClassName="office-wrapper"
            placeholder="Select office"
            selectInputClassName="office-input default-select"
            required
          />
          <Input
            id="roomCapacity"
            name="roomCapacity"
            type="number"
            labelClass="capacity-label"
            inputClass="mrm-input"
            placeholder="0"
            title="Please add numbers only"
            value={roomCapacity}
            labelName="Room Capacity"
            controlsClass="filter-controls"
            onChange={this.handleInputChange}
            required
          />
        </div>
        <div className="filter-footer">
          <button className="footer-btns" onClick={this.handleClear} >Clear Filters</button>
          <button className="footer-btns close" onClick={this.handleClose} >Close</button>
        </div>
      </IconMenu>
    );
  }
}
export default FilterButton;
