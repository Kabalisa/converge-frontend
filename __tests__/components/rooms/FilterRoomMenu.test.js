import React from 'react';
import { shallow } from 'enzyme';

import FilterRoomMenu from '../../../src/components/rooms/FilterRoomMenu';

describe('AddRoomMenu', () => {
  const wrapper = shallow(<FilterRoomMenu />);

  it('should render properly', () => {
    const preventDefault = jest.fn();
    wrapper.instance().handleInputChange({ preventDefault });
    expect(wrapper).toMatchSnapshot();
  });
});
