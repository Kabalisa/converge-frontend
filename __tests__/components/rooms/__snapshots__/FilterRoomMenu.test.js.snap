// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoomMenu should render properly 1`] = `
<ThemedIconMenu
  className="filter-dropdown"
  composeTheme="deeply"
  icon={
    <div
      className="filterBtn"
    >
      <span>
        Filter
      </span>
    </div>
  }
  iconRipple={true}
  mapThemrProps={[Function]}
  menuRipple={true}
  position="auto"
  selectable={false}
>
  <div
    className="filter-search"
  >
    <Input
      controlsClass=""
      id="filterSearch"
      inputClass="mrm-input filter-search-input"
      labelClass="input1"
      labelName=""
      name="filterSearch"
      onChange={[Function]}
      placeholder="Search"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="other-filters"
  >
    <SelectInput
      id="location-filter"
      labelText="Location"
      name="locationFilter"
      onChange={[Function]}
      options={Array []}
      placeholder="Select location"
      placeholderValue=""
      required={true}
      selectInputClassName="location-input default-select"
      value=""
      wrapperClassName="location-wrapper"
    />
    <SelectInput
      id="office-filter"
      labelText="Office"
      name="OfficeFilter"
      onChange={[Function]}
      options={Array []}
      placeholder="Select office"
      placeholderValue=""
      required={true}
      selectInputClassName="office-input default-select"
      value=""
      wrapperClassName="office-wrapper"
    />
    <Input
      controlsClass="filter-controls"
      id="roomCapacity"
      inputClass="mrm-input"
      labelClass="capacity-label"
      labelName="Room Capacity"
      name="roomCapacity"
      onChange={[Function]}
      placeholder="0"
      required={true}
      title="Please add numbers only"
      type="number"
      value={0}
    />
  </div>
  <div
    className="filter-footer"
  >
    <button
      className="footer-btns"
    >
      Clear Filters
    </button>
    <button
      className="footer-btns close"
    >
      Close
    </button>
  </div>
</ThemedIconMenu>
`;
